src/Sentry/Internal/DebugStackTrace.cs

```diff
@@ -382,77 +382,116 @@ private static void DemangleLambdaReturnType(SentryStackFrame frame)
-            _debugImageIndexByModule.Add(id, DebugImageMissing); // don't try to resolve again
+            options.LogDebug("Skipping debug image for module '{0}' because assembly wasn't found: '{1}'",
+                module.Name, assemblyName);
             return null;
         }
 
-        string? codeId = null;
         var headers = peReader.PEHeaders;
-        if (headers.PEHeader is { } peHeader)
-        {
-            codeId = $"{headers.CoffHeader.TimeDateStamp:X8}{peHeader.SizeOfImage:x}";
-        }
+        var codeId = headers.PEHeader is { } peHeader
+            ? $"{headers.CoffHeader.TimeDateStamp:X8}{peHeader.SizeOfImage:x}"
+            : null;
 
         string? debugId = null;
         string? debugFile = null;
         string? debugChecksum = null;
 
-        var debugDirs = peReader.ReadDebugDirectory();
-        foreach (var entry in debugDirs)
+        var debugDirectoryEntries = peReader.ReadDebugDirectory();
+
+        foreach (var entry in debugDirectoryEntries)
         {
-            if (entry.Type == DebugDirectoryEntryType.PdbChecksum)
+            switch (entry.Type)
             {
-                var checksum = peReader.ReadPdbChecksumDebugDirectoryData(entry);
-                var checksumHex = checksum.Checksum.AsSpan().ToHexString();
-                debugChecksum = $"{checksum.AlgorithmName}:{checksumHex}";
+                case DebugDirectoryEntryType.PdbChecksum:
+                {
+                    var checksum = peReader.ReadPdbChecksumDebugDirectoryData(entry);
+                    var checksumHex = checksum.Checksum.AsSpan().ToHexString();
+                    debugChecksum = $"{checksum.AlgorithmName}:{checksumHex}";
+                    break;
+                }
+
+                case DebugDirectoryEntryType.CodeView:
+                {
+                    var codeView = peReader.ReadCodeViewDebugDirectoryData(entry);
+                    debugFile = codeView.Path;
+
+                    // Specification:
+                    // https://github.com/dotnet/runtime/blob/main/docs/design/specs/PE-COFF.md#codeview-debug-directory-entry-type-2
+                    //
+                    // See also:
+                    // https://learn.microsoft.com/dotnet/csharp/language-reference/compiler-options/code-generation#debugtype
+                    //
+                    // Note: Matching PDB ID is stored in the #Pdb stream of the .pdb file.
+
+                    if (entry.IsPortableCodeView)
+                    {
+                        // Portable PDB Format
+                        // Version Major=any, Minor=0x504d
+                        debugId = $"{codeView.Guid}-{entry.Stamp:x8}";
+                    }
+                    else
+                    {
+                        // Full PDB Format (Windows only)
+                        // Version Major=0, Minor=0
+                        debugId = $"{codeView.Guid}-{codeView.Age}";
+                    }
```

Comment:
This is the format we did not previously have.  Everything else in the PR is refactoring and optimization.